lib/matplotlib/tri/triplot.py

```diff
@@ -42,11 +44,11 @@ def triplot(ax, *args, **kwargs):
     linestyle, marker, color = matplotlib.axes._base._process_plot_format(fmt)
 
     # Insert plot format string into a copy of kwargs (kwargs values prevail).
-    kw = kwargs.copy()
+    kw = cbook.normalize_kwargs(kwargs, mlines.Line2D)
     for key, val in zip(('linestyle', 'marker', 'color'),
                         (linestyle, marker, color)):
         if val is not None:
-            kw[key] = kwargs.get(key, val)
+            kw.setdefault(key, val)
```

Comment:
I think this changes precedence if both are given, a linestyle via fmt and a linestyle via kwarg.

Reply:
`kwargs.get(key, val)` gets the value for `key`, falling back to `val` if it doesn't exist, then setting that on the same key in `kw[key]`. `kw.setdefault(key, val)` sets `key` to `val` if it's not already set. They should be the same. Note, `kw` is a copy of `kwargs`.

Reply:
You‘re right.